frontend/views/containers/proposals/ProposalVoteOptions.vue

```diff
@@ -0,0 +1,204 @@
+        return null
+      }
+      try {
+        const proposalHash = this.proposalHash
+        const payload = {}
+        if (oneVoteToPass(proposalHash)) {
+          payload.passPayload = generateInvites(1)
+        }
+        const vote = await sbp('gi.contracts/group/proposalVote/create',
+          {
+            proposalHash,
+            vote: VOTE_FOR,
+            ...payload
+          },
+          this.currentGroupId
+        )
+        await sbp('backend/publishLogEntry', vote)
+
+        if (payload.passPayload) {
+          const groupName = this.groupSettings.groupName
+          const username = this.data.member
+          // TODO: delete this entire section, this is just
+          //       here for debug and testing purposes until
+          //       we get the links page working nicely.
+          //       this.data.members will not contain usernames in the future.
+          const contractID = await sbp('namespace/lookup', username)
+          const identityContract = await sbp('state/latestContractState', contractID)
+          console.debug('sending invite to:', contractID, identityContract)
+          const inviteToMailbox = await sbp('gi.contracts/mailbox/postMessage/create',
+            {
+              messageType: TYPE_MESSAGE,
+              from: groupName,
+              subject: `You've been invited to join ${groupName}!`,
+              message: `Hi ${username},
+              
+              ${groupName} has voted to invite you! Horray!
+              Here's your special invite link:
+              
+              ${buildInvitationUrl(this.$store.state.currentGroupId, payload.passPayload.inviteSecret)}`
+            },
+            identityContract.attributes.mailbox
+          )
+          await sbp('backend/publishLogEntry', inviteToMailbox)
+        }
+      } catch (ex) {
+        console.log(ex)
+        this.ephemeral.errorMsg = L('Failed to Cast Vote. Try again.')
+      }
+    },
+    async voteAgainst () {
+      this.ephemeral.changingVote = false
+      this.ephemeral.errorMsg = null
+      const isSure = confirm(L('Are you sure you want to vote no?'))
+      // Avoid redundant vote from "Change vote" if already voted AGAINST before
+      if (!isSure || this.proposal.votes[this.currentUsername] === VOTE_AGAINST) {
+        return null
+      }
+      try {
+        const vote = await sbp('gi.contracts/group/proposalVote/create',
+          {
+            proposalHash: this.proposalHash,
+            vote: VOTE_AGAINST
+          },
+          this.currentGroupId
+        )
+        await sbp('backend/publishLogEntry', vote)
+      } catch (ex) {
+        console.log(ex)
+        this.ephemeral.errorMsg = L('Failed to Cast Vote. Try again.')
+      }
+    },
+    async cancelProposal () {
+      const isSure = confirm(L('Are you sure you want to cancel this proposal?'))
+      if (!isSure) {
+        return null
+      }
+      try {
+        const proposalHash = this.proposalHash
```

Comment:
In general (unless it really helps with readability), if you only use a variable once you should inline it to save lines

Reply:
here it helps for readability and to assign directly the object key